clickstream/src/main/java/software/aws/solution/clickstream/client/AnalyticsEvent.java

```diff
@@ -509,6 +510,7 @@ public JSONObject toJSONObject() {
         builder.withAttribute("app_title", this.appDetails.getAppTitle());
         builder.withAttribute("user", this.userAttributes);
         builder.withAttribute("attributes", this.attributes);
+        builder.withAttribute("hashCode", StringUtil.getHashCode(builder.toJSONObject().toString()));
```

Comment:
This line will not change the original key order, because `hashCode` will be added first when initializing the builder object, see [JSONBuilder](https://github.com/awslabs/clickstream-android/blob/feature_hash_code/clickstream/src/main/java/software/aws/solution/clickstream/client/util/JSONBuilder.java#L40)

The key sequence is:
```
{
    "hashCode": "80452b0",
    ...
    "user": {...},
    "attributes": {...}
}

Reply:
That is my concern. There are three JSON builders in different language implementation. They might output the same event object with different JSON strings which are the inputs of hash algorithm.

Did you check the same event object generated same serialized JSON string in different implementation?

Reply:
Because each SDK will have specific fields, the actual generated JSON content will vary.

Assuming that only common fields are generated and the Event object attributes and values are all the same, js and java will be serialized in the order in which the fields are added. However, swift does not support this feature and can only fix the order of json fields in alphabetical order. 

But this does not affect calculating the hashCode of each event and determining whether it has been changed, as verified in the test case. And since they all use the `SHA256` algorithm, for the same json string, the three SDKs will produce consistent hashCode after testing.